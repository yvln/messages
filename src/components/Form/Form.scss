@import "styles/_variables.scss";

.Form {
  background-color: $color-lightest;
  padding: $spacing-medium;

  > div {
    color: $color;
    font-weight: 600;
    margin-bottom: $spacing-small;
  }

  .error {
    position: absolute;
    font-size: 12px;
    font-weight: 600;
  }
}

.Form-clickable {
  display: flex;
  justify-content: flex-end;
}