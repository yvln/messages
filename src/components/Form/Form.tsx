import * as React from 'react';

import Button from '../Button';
import Container from '../Container';
import Select from '../Select';
import TextArea from '../TextArea';
import './Form.scss';

export interface IStateProps {
  isFailure: boolean;
}

export interface IDispatchProps {
  fetchMessages: () => void;
  postMessage: (message: string, isPrivate: boolean, username: string) => void;
}

export interface IInjectedProps {
  username?: string;
}

interface IState {
  message: string;
  private: boolean;
}

export type IProps = IStateProps & IDispatchProps & IInjectedProps;

class Form extends React.Component<IProps, IState> {
  state = {
    message: '',
    private: false,
  };

  componentDidMount() {
    // @ts-ignore
    document.addEventListener('keydown', this.keydownHandler);
  }

  componentWillUnmount() {
    // @ts-ignore
    document.removeEventListener('keydown', this.keydownHandler);
  }

  keydownHandler = (event: React.KeyboardEvent): any => {
    if (event.keyCode === 13 && event.ctrlKey) {
      this.handleSubmit();
    }
  };

  handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
    this.setState({
      private: event.target.value === 'private',
    });
  };

  handleChangeTextArea = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
    this.setState({
      message: event.target.value,
    });
  };

  handleSubmit = async (
    event?: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    if (event) {
      event.preventDefault();
    }

    const username = this.props.username;
    if (!username) {
      return;
    }

    if (!this.state.message.trim()) {
      return;
    }

    try {
      await this.props.postMessage(
        this.state.message,
        this.state.private,
        username
      );
      this.resetForm();
      this.props.fetchMessages();
      // tslint:disable-next-line
    } catch (err) {}
  };

  resetForm = () => {
    this.setState({
      message: '',
      private: false,
    });
  };

  render() {
    return (
      <Container>
        <div className="Form">
          <div>{this.props.username}</div>
          <form>
            <TextArea
              onChange={this.handleChangeTextArea}
              value={this.state.message}
            />
            {this.props.isFailure && (
              <div className="error">
                We were unable to post your message. Please try again later.
              </div>
            )}
            <div className="Form-clickable">
              <Select onChange={this.handleChangeSelect} />
              <Button onClick={this.handleSubmit}>
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  width="15"
                  height="15"
                  viewBox="0 0 24 24"
                >
                  <path d="M8.122 24l-4.122-4 8-8-8-8 4.122-4 11.878 12z" />
                </svg>
              </Button>
            </div>
          </form>
        </div>
      </Container>
    );
  }
}

export default Form;
