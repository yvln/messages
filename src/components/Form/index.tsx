import { connect } from 'react-redux';

import * as actions from '../../modules/actions/messages';
import { Dispatch, IFullState } from '../../types';
import Form, { IDispatchProps, IStateProps } from './Form';

const mapStateToProps = (state: IFullState): IStateProps => ({
  isFailure: state.messages.post.isFailure,
})

const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  return {
    fetchMessages: () => {
      dispatch(actions.fetchMessages());
    },
    postMessage: (message: string, isPrivate: boolean, username: string) => {
      dispatch(actions.postMessage(message, isPrivate, username));
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Form);
