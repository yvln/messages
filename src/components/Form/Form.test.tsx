import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import Form, { IProps } from './Form';

configure({ adapter: new Adapter() });

describe('src/components/Form', () => {
  let props: IProps;

  beforeEach(() => {
    props = {
      fetchMessages: jest.fn(),
      isFailure: false,
      postMessage: jest.fn(),
      username: 'yvln',
    };
  });

  it('should render', () => {
    const component = shallow(<Form {...props} />);
    expect(component).toMatchSnapshot();
  });
});
