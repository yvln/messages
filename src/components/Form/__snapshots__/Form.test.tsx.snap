// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Form should display an error if  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    fetchMessages={[MockFunction]}
    isFailure={false}
    postMessage={[MockFunction]}
    username="yvln"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="Form"
      >
        <div>
          yvln
        </div>
        <form>
          <TextArea
            onChange={[Function]}
            onKeyDown={[Function]}
            value=""
          />
          <div
            className="Form-clickable"
          >
            <Select
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "label": "Public",
                    "value": "public",
                  },
                  Object {
                    "label": "Private",
                    "value": "private",
                  },
                ]
              }
            />
            <Button
              onClick={[Function]}
            >
              <Svg
                height={15}
                name="arrowRight"
                width={15}
              />
            </Button>
          </div>
        </form>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            yvln
          </div>,
          <form>
            <TextArea
              onChange={[Function]}
              onKeyDown={[Function]}
              value=""
            />
            <div
              className="Form-clickable"
            >
              <Select
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Public",
                      "value": "public",
                    },
                    Object {
                      "label": "Private",
                      "value": "private",
                    },
                  ]
                }
              />
              <Button
                onClick={[Function]}
              >
                <Svg
                  height={15}
                  name="arrowRight"
                  width={15}
                />
              </Button>
            </div>
          </form>,
        ],
        "className": "Form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "yvln",
          },
          "ref": null,
          "rendered": "yvln",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextArea
                onChange={[Function]}
                onKeyDown={[Function]}
                value=""
              />,
              false,
              <div
                className="Form-clickable"
              >
                <Select
                  onChange={[Function]}
                  options={
                    Array [
                      Object {
                        "label": "Public",
                        "value": "public",
                      },
                      Object {
                        "label": "Private",
                        "value": "private",
                      },
                    ]
                  }
                />
                <Button
                  onClick={[Function]}
                >
                  <Svg
                    height={15}
                    name="arrowRight"
                    width={15}
                  />
                </Button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onChange": [Function],
                "onKeyDown": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Select
                    onChange={[Function]}
                    options={
                      Array [
                        Object {
                          "label": "Public",
                          "value": "public",
                        },
                        Object {
                          "label": "Private",
                          "value": "private",
                        },
                      ]
                    }
                  />,
                  <Button
                    onClick={[Function]}
                  >
                    <Svg
                      height={15}
                      name="arrowRight"
                      width={15}
                    />
                  </Button>,
                ],
                "className": "Form-clickable",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onChange": [Function],
                    "options": Array [
                      Object {
                        "label": "Public",
                        "value": "public",
                      },
                      Object {
                        "label": "Private",
                        "value": "private",
                      },
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <Svg
                      height={15}
                      name="arrowRight"
                      width={15}
                    />,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "height": 15,
                      "name": "arrowRight",
                      "width": 15,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="Form"
        >
          <div>
            yvln
          </div>
          <form>
            <TextArea
              onChange={[Function]}
              onKeyDown={[Function]}
              value=""
            />
            <div
              className="Form-clickable"
            >
              <Select
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Public",
                      "value": "public",
                    },
                    Object {
                      "label": "Private",
                      "value": "private",
                    },
                  ]
                }
              />
              <Button
                onClick={[Function]}
              >
                <Svg
                  height={15}
                  name="arrowRight"
                  width={15}
                />
              </Button>
            </div>
          </form>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              yvln
            </div>,
            <form>
              <TextArea
                onChange={[Function]}
                onKeyDown={[Function]}
                value=""
              />
              <div
                className="Form-clickable"
              >
                <Select
                  onChange={[Function]}
                  options={
                    Array [
                      Object {
                        "label": "Public",
                        "value": "public",
                      },
                      Object {
                        "label": "Private",
                        "value": "private",
                      },
                    ]
                  }
                />
                <Button
                  onClick={[Function]}
                >
                  <Svg
                    height={15}
                    name="arrowRight"
                    width={15}
                  />
                </Button>
              </div>
            </form>,
          ],
          "className": "Form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "yvln",
            },
            "ref": null,
            "rendered": "yvln",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TextArea
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />,
                false,
                <div
                  className="Form-clickable"
                >
                  <Select
                    onChange={[Function]}
                    options={
                      Array [
                        Object {
                          "label": "Public",
                          "value": "public",
                        },
                        Object {
                          "label": "Private",
                          "value": "private",
                        },
                      ]
                    }
                  />
                  <Button
                    onClick={[Function]}
                  >
                    <Svg
                      height={15}
                      name="arrowRight"
                      width={15}
                    />
                  </Button>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "onChange": [Function],
                  "onKeyDown": [Function],
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Select
                      onChange={[Function]}
                      options={
                        Array [
                          Object {
                            "label": "Public",
                            "value": "public",
                          },
                          Object {
                            "label": "Private",
                            "value": "private",
                          },
                        ]
                      }
                    />,
                    <Button
                      onClick={[Function]}
                    >
                      <Svg
                        height={15}
                        name="arrowRight"
                        width={15}
                      />
                    </Button>,
                  ],
                  "className": "Form-clickable",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "onChange": [Function],
                      "options": Array [
                        Object {
                          "label": "Public",
                          "value": "public",
                        },
                        Object {
                          "label": "Private",
                          "value": "private",
                        },
                      ],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <Svg
                        height={15}
                        name="arrowRight"
                        width={15}
                      />,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "height": 15,
                        "name": "arrowRight",
                        "width": 15,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`src/components/Form should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    fetchMessages={[MockFunction]}
    isFailure={false}
    postMessage={[MockFunction]}
    username="yvln"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="Form"
      >
        <div>
          yvln
        </div>
        <form>
          <TextArea
            onChange={[Function]}
            onKeyDown={[Function]}
            value=""
          />
          <div
            className="Form-clickable"
          >
            <Select
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "label": "Public",
                    "value": "public",
                  },
                  Object {
                    "label": "Private",
                    "value": "private",
                  },
                ]
              }
            />
            <Button
              onClick={[Function]}
            >
              <Svg
                height={15}
                name="arrowRight"
                width={15}
              />
            </Button>
          </div>
        </form>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            yvln
          </div>,
          <form>
            <TextArea
              onChange={[Function]}
              onKeyDown={[Function]}
              value=""
            />
            <div
              className="Form-clickable"
            >
              <Select
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Public",
                      "value": "public",
                    },
                    Object {
                      "label": "Private",
                      "value": "private",
                    },
                  ]
                }
              />
              <Button
                onClick={[Function]}
              >
                <Svg
                  height={15}
                  name="arrowRight"
                  width={15}
                />
              </Button>
            </div>
          </form>,
        ],
        "className": "Form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "yvln",
          },
          "ref": null,
          "rendered": "yvln",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextArea
                onChange={[Function]}
                onKeyDown={[Function]}
                value=""
              />,
              false,
              <div
                className="Form-clickable"
              >
                <Select
                  onChange={[Function]}
                  options={
                    Array [
                      Object {
                        "label": "Public",
                        "value": "public",
                      },
                      Object {
                        "label": "Private",
                        "value": "private",
                      },
                    ]
                  }
                />
                <Button
                  onClick={[Function]}
                >
                  <Svg
                    height={15}
                    name="arrowRight"
                    width={15}
                  />
                </Button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onChange": [Function],
                "onKeyDown": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Select
                    onChange={[Function]}
                    options={
                      Array [
                        Object {
                          "label": "Public",
                          "value": "public",
                        },
                        Object {
                          "label": "Private",
                          "value": "private",
                        },
                      ]
                    }
                  />,
                  <Button
                    onClick={[Function]}
                  >
                    <Svg
                      height={15}
                      name="arrowRight"
                      width={15}
                    />
                  </Button>,
                ],
                "className": "Form-clickable",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onChange": [Function],
                    "options": Array [
                      Object {
                        "label": "Public",
                        "value": "public",
                      },
                      Object {
                        "label": "Private",
                        "value": "private",
                      },
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <Svg
                      height={15}
                      name="arrowRight"
                      width={15}
                    />,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "height": 15,
                      "name": "arrowRight",
                      "width": 15,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="Form"
        >
          <div>
            yvln
          </div>
          <form>
            <TextArea
              onChange={[Function]}
              onKeyDown={[Function]}
              value=""
            />
            <div
              className="Form-clickable"
            >
              <Select
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Public",
                      "value": "public",
                    },
                    Object {
                      "label": "Private",
                      "value": "private",
                    },
                  ]
                }
              />
              <Button
                onClick={[Function]}
              >
                <Svg
                  height={15}
                  name="arrowRight"
                  width={15}
                />
              </Button>
            </div>
          </form>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              yvln
            </div>,
            <form>
              <TextArea
                onChange={[Function]}
                onKeyDown={[Function]}
                value=""
              />
              <div
                className="Form-clickable"
              >
                <Select
                  onChange={[Function]}
                  options={
                    Array [
                      Object {
                        "label": "Public",
                        "value": "public",
                      },
                      Object {
                        "label": "Private",
                        "value": "private",
                      },
                    ]
                  }
                />
                <Button
                  onClick={[Function]}
                >
                  <Svg
                    height={15}
                    name="arrowRight"
                    width={15}
                  />
                </Button>
              </div>
            </form>,
          ],
          "className": "Form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "yvln",
            },
            "ref": null,
            "rendered": "yvln",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TextArea
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />,
                false,
                <div
                  className="Form-clickable"
                >
                  <Select
                    onChange={[Function]}
                    options={
                      Array [
                        Object {
                          "label": "Public",
                          "value": "public",
                        },
                        Object {
                          "label": "Private",
                          "value": "private",
                        },
                      ]
                    }
                  />
                  <Button
                    onClick={[Function]}
                  >
                    <Svg
                      height={15}
                      name="arrowRight"
                      width={15}
                    />
                  </Button>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "onChange": [Function],
                  "onKeyDown": [Function],
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Select
                      onChange={[Function]}
                      options={
                        Array [
                          Object {
                            "label": "Public",
                            "value": "public",
                          },
                          Object {
                            "label": "Private",
                            "value": "private",
                          },
                        ]
                      }
                    />,
                    <Button
                      onClick={[Function]}
                    >
                      <Svg
                        height={15}
                        name="arrowRight"
                        width={15}
                      />
                    </Button>,
                  ],
                  "className": "Form-clickable",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "onChange": [Function],
                      "options": Array [
                        Object {
                          "label": "Public",
                          "value": "public",
                        },
                        Object {
                          "label": "Private",
                          "value": "private",
                        },
                      ],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <Svg
                        height={15}
                        name="arrowRight"
                        width={15}
                      />,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "height": 15,
                        "name": "arrowRight",
                        "width": 15,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
